mlflow/models/evaluation/utils/metric.py

```diff
@@ -0,0 +1,125 @@
+import logging
+from dataclasses import dataclass
+from typing import Callable, Dict, Optional
+
+import numpy as np
+
+from mlflow.metrics.base import MetricValue
+from mlflow.models.evaluation.base import EvaluationMetric
+
+_logger = logging.getLogger(__name__)
+
+
+@dataclass
+class MetricDefinition:
+    """
+    A namedtuple representing a metric function and its properties.
+
+    function : the metric function
+    name : the name of the metric function
+    index : the index of the function in the ``extra_metrics`` argument of mlflow.evaluate
+    """
+
+    function: Callable
```

Comment:
```suggestion
    function: Callable[..., Any]
```

to make this less ambiguous.

Reply:
oh didn't know `...` is a valid argument for type hints!